### DISCUSSION TITLE
User/Guest account creation for domain-specific monitoring

### DISCUSSION THREAD

**Discussion by nu-ad** [towfiqi/serpbear#152] [Ideas]
Hi!

I am writing to suggest a potential feature enhancement for the SERPBEAR tool that I believe could significantly augment its functionality and user experience.

I have identified a potential area where the tool could be expanded: the creation of guest or user accounts with limited access privileges. The core idea is to allow these accounts to log in and monitor specific domains assigned to them, without having access to broader functionalities or sensitive data like in admin account case.

Proposed feature overview:

- Account type: Creation of guest/user accounts alongside the primary admin account.
- Access control: These accounts would have restricted access, limited to monitoring certain predefined domains.
- Functionality: Users can view and monitor the SERP performance of the assigned domains but cannot modify settings, access other domains, or utilize other tool features beyond their scope.
- Security and privacy: Ensuring these restricted accounts cannot access or alter critical settings or sensitive data.

This feature could be highly beneficial for agencies or teams where multiple users need to monitor different domains without requiring full access to the tool. It could also enhance security by limiting access based on user roles and responsibilities.

I suspect further expansion of the DB usage and functionality would be required in order to achieve this. As well as major adjustemnts in SERPBEAR's core parts. 

Would such a feature be feasible to implement within the existing framework of SERPBEAR? If so, could you provide some insights into the potential timeline and any technical considerations that might be involved?

Thanks for considering this suggestion. I believe this feature could add significant value to SERPBEAR, making it even more versatile and user-friendly for diverse user groups.

Looking forward to your response and any feedback you may have.

Regards,
J

**Comment by jchambo** [comment]
This is 100% a feature that is needed and I'm surprised it's not in the tool out of the box or has been implemented since this post.

@towfiqi - any response?
